test/neural_de.jl

```diff
@@ -244,8 +334,21 @@ grads = Zygote.gradient(()->sum(sode(x)),Flux.params(x,sode))
 @test ! iszero(grads[sode.p])
 @test ! iszero(grads[sode.p][end])
 
+sode = NeuralDSDE(fastcdudt,fastcdudt2,(0.0f0,.1f0),SOSRI(),saveat=0.0:0.01:0.1)

```

Comment:
if you set RNG seeds, do these gradients match the uncached versions? That's a good test.